libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageFragment.kt

```diff
@@ -119,10 +157,17 @@ class PreviewImageFragment : Fragment() {
     }
 
     private fun navigateToCropScreenWithInputFilePath(fileInfo: Pair<String, String?>) {
-        val inputFilePath = fileInfo.first
-        val outputFileExtension = fileInfo.second
-        findNavController().navigate(
-            PreviewImageFragmentDirections.actionPreviewFragmentToCropFragment(inputFilePath, outputFileExtension)
-        )
+        // TODO: temporarily stop navigation to next screen
+//        val inputFilePath = fileInfo.first
+//        val outputFileExtension = fileInfo.second
+//        findNavController().navigate(
+//            PreviewImageFragmentDirections.actionPreviewFragmentToCropFragment(inputFilePath, outputFileExtension)
+//        )
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        previewImageViewPager.adapter?.unregisterAdapterDataObserver(pagerAdapterObserver)
```

Comment:
I'm just wondering if we can set these fields simply to `null` -> they are no longer valid? AFAIK gc will be able to collect both the fields and the views. Wdyt?

Reply:
`previewImageViewPager.adapter` is set to null inside `TabLayoutMediator` `detach()` called in the following line. I've also set `pagerAdapterObserver` to `null` in ea85194. Hope this is what you meant by 

> these fields

Reply:
Sorry, my comment wasn't clear. What I meant is if we can set both the fields to null instead of calling "deatch" and "unregisterAdapterDataObserver". It seems weird that we'd need to manually unregister from a UI component - or have you find this information somewhere in the docs? Wdyt?

Reply:
So I checked `TabLayoutMediator` source code and observed inside `detach()` it [unregisters the adapter observer](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/tabs/TabLayoutMediator.java#L150-L153) if `autoRefresh` is enabled. Since we haven't enabled `autoRefresh`, I considered unregistering it explicitly in an appropriate lifecycle method:

`onViewCreated -> registerAdapterDataObserver`
`onDestroyView -> unregisterAdapterDataObserver`

I guess there's no harm in doing so?

Reply:
Ok, lets keep it. It makes the code more complicated, but I guess it's better to play it safe in this case. Thanks for looking into it.